korman/properties/modifiers/logic.py

```diff
@@ -117,3 +117,122 @@ def export(self, exporter, bo, so):
     @property
     def requires_actor(self):
         return True
+
+
+telescope_pfm = {
+    "filename": "xTelescope.py",
+    "attribs": (
+        { 'id':  1, 'type': "ptAttribActivator", 'name': "Activate" },
+        { 'id':  2, 'type': "ptAttribSceneobject", 'name': "Camera" },
+        { 'id':  3, 'type': "ptAttribBehavior", 'name': "Behavior" },
+        { 'id':  4, 'type': "ptAttribString", 'name': "Vignette" },
+    )
+}
+
+
+class PlasmaTelescope(PlasmaModifierProperties, PlasmaModifierLogicWiz):
+    pl_id="telescope"
+
+    bl_category = "Logic"
+    bl_label = "Telescope"
+    bl_description = "Set up clickable mesh as a telescope."
+    bl_icon = "VISIBLE_IPO_ON"
+
+    clickable_object = PointerProperty(name="Clickable",
+                                       description="Clickable object for telescope.",
+                                       type=bpy.types.Object,
+                                       poll=idprops.poll_mesh_objects)
+    region_object = PointerProperty(name="Click Region",
+                                    description="Area where clickable becomes active.",
+                                    type=bpy.types.Object,
+                                    poll=idprops.poll_mesh_objects)
+    oneshot_object = PointerProperty(name="OneShot",
+                                     description="Empty object used to orientate avatar in front of telescope.",
+                                     type=bpy.types.Object,
+                                     poll=idprops.poll_empty_objects)
+    camera_object = PointerProperty(name="Camera",
+                                    description="Camera used when viewing through telescope.",
+                                    type=bpy.types.Object,
+                                    poll=idprops.poll_camera_objects)
+    telescope_name = StringProperty(name="Name (Optional)",
+                                    description="Name for your telescope (optional).",
+                                    options=set())
```

Comment:
From what I can tell, this is the name of the GUI dialog to use for the telescope (rather than the name of the telescope itself, which is what the description above implies). Since we don't have the ability to export GUIs at the moment, I am tempted to say that this should be removed until we have some scaffolding around GUI dialogs, then we'll be able to make a more informed decision on how to expose this setting. For now, I think the value for `Vignette` in the PythonFileMod should be hardwired to `telescope`, which is the default global GUI for most telescopes in the game.

Reply:
there are at least two different telescope options, could hardwire it for either

Reply:
Can you clarify the name of the other option? I can't remember the name or what it looks like off the top of my head.

Reply:
me either, ill look into it with some prp snooping

Reply:
After looking through for a while, I cant find any others that use the xTelescope.py.
I found other telescope type stuff but they have custom scripts.
Everything uses `telescope` for xTelescope.py